Automattic/woocommerce-services
# Keep live rates enabled even with WC Shipping active

## Description


Context for development decisions: p1726023174650669-slack-C04KWSNPSE5

This PR keeps live rates enabled even when both WCS&T and WCShip are active. It does this by bringing back the registration of live rates shipping method registration, as well as bringing back the WCS&T settings page (hidden but available through the shipping method settings and in the future, also [through the WCShipping settings page](https://github.com/woocommerce/woocommerce-shipping/pull/748)).

Additionally, it uses package data from WCShipping if a store has been migrated from WCS&T to WCShipping. Since the two extensions use different package data formats, this PR introduces remapping of package data formats on both reads and writes to `wc_connect_options`.

## Breaking changes

The changes described below do not affect how WCShipping stores data, only how WCS&T remaps WCShipping's package data for its own purposes.

However, since WCS&T is writing data to WCShipping's options field when an update is requested from the WCS&T settings page that this PR reinstates for package management, it does affect the values that end up in `wcshipping_options[packages]`.

To clean up the package data in WCShipping (not a required step; just for the sake of clarity), separate PRs will have to be implemented there.

### 1) Outer dimensions of custom packages

This is a feature that seems long deprecated but still supported in the codebase.

In WCS&T, predefined packages have outer dimensions (dimensions of the box used for calculating shipping rates) and inner dimensions (dimensions of the inside of the box used for box packing cart contents).

The WCS&T `wc_connect_options` option allows the key "outer_dimensions" to be defined on custom packages too. This [used to be a feature of the extension](https://github.com/Automattic/woocommerce-services/issues/325) but [it was removed in Sep 2019](https://github.com/Automattic/woocommerce-services/pull/1672).

Although it is no longer possible to set outer dimensions in WCS&T, the field was never unset if a custom package had it set (AFAIK). This means that some stores might have this key lying around for some old custom packages, although it is not presented anywhere anymore.

We could continue passing this key around however since the package edit modal in both WCS&T and WCShipping does not allow you to edit outer dimensions - it only supports inner dimensions (labelled simply as "Dimensions") - there is no way to edit them. This means that merchants effectively have no way to edit the dimensions used for calculating live rates for some very old packages, if they have any still around.

Because of the above, I decided to not support this field in this PR and to only carry over the `inner_dimensions` field which is the one that merchants can edit (and is the only one set for custom packages created after Sep 2019).

### 2) Custom package type value stored under `type`/`isLetter`/`is_letter`

WCS&T used to define packages as a boolean `is_letter`. If a package is a letter, it means it's an envelope. Otherwise, it's a box.

WCShipping introduces a new field called `type` which can have the value of `box` or `envelope`. To support old packages, the field `is_letter` is still kept around, as `isLetter`. In addition to that, a package might also have `is_letter` set.

Since it seems these are generally in sync for the packages defined by WCShipping, I have opted to only keep the `type` field after consulting with the team (p1726489562638679-slack-C04KWSNPSE5).

### Testing instructions

Testing live rates pre-migration:

1. Install WCS&T on a non-migrated-to-WCShipping site (or reset the WCShip migration wizard by running the following code):
   ```
   delete_option( 'wcshipping_migration_state' );
   delete_option( 'wcst_data_migration_required' );
   delete_option( 'wcshipping_installation_completed_shown' );
   delete_option( 'wcst_data_migration_processes_to_run' );
   delete_option( 'wcshipping_version' );
   ```
3. Add some packages in WC > Settings > WooCommerce Shipping.
4. Ensure your site is eligible for USPS live rates.
5. Add a shipping zone with a live rates shipping method.
6. Verify you can access and edit the shipping method's settings.
7. Install and activate WooCommerce Shipping.
8. Verify that live rates are working on the frontend.
   - Please check both the blocks and the classic version of the cart and checkout pages.
9. Verify that label purchases are working.
   - The label purchase modal should display the appropriate shipping costs based on the selected live rates shipping method.
10. Verify the status pages are working for both WCS&T and WCShip.
11. Verify plugin settings are working for both WCS&T and WCShip.

Testing live rates post-migration (after going through all of the above tests):

1. Migrate the settings from WCS&T to WCShip.
2. Go to `admin.php?page=wc-settings&tab=shipping&section=woocommerce-services-settings` and verify you can see the packages you previously had in WCS&T.
3. Add packages of different dimensions/weight/types on the new live rates settings page (WC > Settings > Shipping > WooCommerce Shipping > Live rates > Select packages). Verify it shows up correctly in the "purchase label" modal.
4. Add packages of different dimensions/weight/types in the "purchase label" modal. Verify it shows up correctly in the live rates settings.
5. Verify that live rates are working on the frontend.
   - Please check both the blocks and the classic version of the cart and checkout pages.
9. Verify that label purchases are working.
   - The label purchase modal should display the appropriate shipping costs based on the selected live rates shipping method.
10. Verify the status pages are working for both WCS&T and WCShip.
11. Verify plugin settings are working for both WCS&T and WCShip.

Testing WCShipping deactivation (after going through all of the above tests):

1. Verify that after deactivating WCShipping, WCS&T displays the packages added/modified/removed during your testing.

### Checklist
